### DISCUSSION TITLE
How to add plugins in docker?

### DISCUSSION THREAD

**Discussion by yjiawqgj** [Flexget/Flexget#3717] [General]
The wiki only writes about adding plugins to the underlying system, without any docker information.

**Comment by ksurl** [comment]
https://flexget.com/InstallWizard/Docker#custom-entrypoint

Need to create a custom script for the entrypoint.

**Reply by yjiawqgj** [reply]
Due to the problem of Google translation, the solution you gave me, as I understand it, means that if you configure the "entrypoint.sh" file and load it into docker's flexget, you can directly write the yaml file, that's it? That is "entrypoint.sh" contains all plugins?

**Reply by ksurl** [reply]
the  yaml file is different from entrypoint. you can change yaml file any where not only in docker. and for any plugin you want to add, you can have it installed with entrypoint.sh before flexget starts. just need to add the command before the last command for startup

**Reply by yjiawqgj** [reply]
I am a novice, and I am learning flexget. I understand that yaml is a task file, and entrypoint.sh is an installation plug-in. I probably understand the meaning of the content of entrypoint.sh. The first line should say add a plug-in? Or a plugin library? The second line should be to install the plugin. But I am a novice, how do I use entrypoint.sh, for example, I want to install rss and transmission plug-ins, can you write an example for me? The official wiki only has the usage method of each plug-in in yaml, but there is no installation method.

**Reply by ksurl** [reply]
For rss and transmission it is already included

**Reply by yjiawqgj** [reply]
Are all the plugins listed in the official wiki included? If the DOCKER does not contain the plugins I need, how can I use entrypoint.sh to add the plugins I need?

**Reply by ksurl** [reply]
Docker version includes all plugins for flexget in the main version. Basically any plugin for flexget in the wiki is already in installed with flexget because it is part of flexget. What you are getting confused on is some plugins require packages to be installed. Like transmission requires transmission-rpc (which is already included in docker version). But need to install if running manually with pip install
